packages/webpack/src/config/base.js

```diff
@@ -80,24 +80,22 @@ export default class WebpackBaseConfig {
   }
 
   getBabelOptions () {
+    const envName = this.name
     const options = {
       ...this.buildContext.buildOptions.babel,
-      envName: this.name
+      envName
     }
 
     if (options.configFile !== false) {
       return options
     }
 
-    const defaultPreset = [
-      require.resolve('@nuxt/babel-preset-app'),
-      {
-        buildTarget: this.isServer ? 'server' : 'client'
-      }
-    ]
+    const defaultPreset = [ require.resolve('@nuxt/babel-preset-app'), {} ]
 
     if (typeof options.presets === 'function') {
-      options.presets = options.presets({ isServer: this.isServer }, defaultPreset)
+      // TODO: remove isServer in Nuxt 3
```

Comment:
Some may prefer this convention (as it is also in other ideas) so maybe worth having `isServer`/`isClient`/`isModern` with nuxt 3 or deprecate them everywhere?

Reply:
My thought was that `isServer` in current babel is for user to differ babel config in different builds(server/client), with this pr, user doesn't need to check it in most cases since `preset-app` has done it automatically via `envName`.

You comment also makes sense, we can add `isClient, isModern`, but do we have envName as arg as well ?

Reply:
Don't know. That's totally something optional and personally even i like the new `envName` more! Tought if it does not add much complexity or perf concerns, we can have both :)

Reply:
Cool, I'll add them all

Reply:
Added, I'll update readme and doc tonight.